FlowCrypt/Core/Models/KeyDetails.swift

```diff
@@ -40,6 +40,15 @@ extension KeyDetails {
     var pgpUserEmails: [String] {
         users.map { MCOAddress(nonEncodedRFC822String: $0).mailbox }
     }
+    
+    var isKeyUsuable: Bool {
+        var expired = false
+        let now = Date()
+        if let expiration = expiration, TimeInterval(expiration) < now.timeIntervalSince1970 {
+            expired = true
+        }
+        return !revoked && !expired
+    }
 }
```

Comment:
This is not ideal, but I see our current js response is missing this. We have to improve this after #1362 

Since this is only a rough approximation

Reply:
https://github.com/FlowCrypt/flowcrypt-ios/issues/1546